### PR TITLE
adding support for mkdir, and mkdirp in the file service

### PR DESCRIPTION
**Pull request by TonyRice**
See issue #397

### CODE REVIEW

File: asyncy/processing/internal/File.py [hunk 1]
```diff
@@ -24,6 +25,36 @@ def safe_path(story, path):
     return f'{story.get_tmp_dir()}{os.fspath(path)}'
 
 
+def mkdir_error(e: IOError, path, story, line):
+    if e.errno == errno.EEXIST and os.path.isdir(path):
```
Comment (wilzbach) [inline asyncy/processing/internal/File.py]: Use the `exist_ok` instead ;-)

File: asyncy/processing/internal/File.py [hunk 2]
```diff
@@ -24,6 +25,36 @@ def safe_path(story, path):
...
+    path = safe_path(story, resolved_args['path'])
+    try:
+        os.mkdir(path)
+    except IOError as e:
+        mkdir_error(e, path, story, line)
+
+
+@Decorators.create_service(name='file', command='mkdirp', arguments={
+    'path': {'type': 'string'}
+})
```
Comment (wilzbach) [inline asyncy/processing/internal/File.py]: We want to abstract complexity for the user. This shouldn't be a separate method. I think we should do this by default.

Comment (TonyRice) [reply asyncy/processing/internal/File.py]: That's what I was initially thinking, I'll go ahead and do that

File: asyncy/processing/internal/File.py (hunk 1 above)
Comment (TonyRice) [reply asyncy/processing/internal/File.py]: Good thinking. Makes life easier.

File: asyncy/processing/internal/File.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+import errno
```
Comment (wilzbach) [inline asyncy/processing/internal/File.py]: Unused

Comment (TonyRice) [reply asyncy/processing/internal/File.py]: Removed!

**Pull request merged**